ports/outcome/portfile.cmake

```
@@ -1,13 +1,55 @@
-# header-only library
-vcpkg_from_github(
-    OUT_SOURCE_PATH SOURCE_PATH
-    REPO ned14/outcome
-    REF 34f3bd55e2bcaf246cb79efe64a5600e89b91b66 #v2.1.3
-    SHA512 83eba50e2095e7c768dacb3af5f82db117c3451f1d5bc2f73d716608d56f7b73006ec33d0f3842fdefd076f0e82b72ece5777868712f75e83eac93aa8adf351c
-    HEAD_REF develop
+set(OUTCOME_VERSION v2.2.0)
+set(OUTCOME_GITCOMMIT ac552d1c69ef556a1327393a0c56092517ff92db)
+set(OUTCOME_HASH 2d27434bcb2d5bc27015fedc221eafd221acb7c873277c4f0c78bc6aa657e6e962f0458e64b2de5f695600a5b15559cd0d3602f276f35c414dfafb34c2214e8c)
+
+# Use Outcome's all sources tarball, not its github repo. The tarball includes all dependencies.
+vcpkg_download_distfile(
+  ALL_SOURCES_TARBALL
+  URLS "https://github.com/ned14/outcome/releases/download/all_tests_passed_${OUTCOME_GITCOMMIT}/outcome-v2-all-sources-${OUTCOME_GITCOMMIT}.tar.xz"
+  FILENAME "outcome-v2-all-sources-${OUTCOME_GITCOMMIT}.tar.xz"
+  SHA512 ${OUTCOME_HASH}
+)
+vcpkg_extract_source_archive_ex(
+  SKIP_PATCH_CHECK
+  OUT_SOURCE_PATH SOURCE_PATH
+  ARCHIVE "${ALL_SOURCES_TARBALL}"
+  REF "outcome-${OUTCOME_GITCOMMIT}"
+)
```

```suggestion
vcpkg_from_github(
    OUT_SOURCE_PATH SOURCE_PATH
    REPO ned14/outcome
    REF all_tests_passed_${OUTCOME_GITCOMMIT}
    SHA512 ${OUTCOME_HASH}
    HEAD_REF develop
)
```

Can we use `vcpkg_from_github` instead?

---

We cannot. The all sources tarball contains all the dependencies, and more importantly, the precise versions of those for which Outcome's CI pass succeeded.

---

Which quickcpplib version (or submission ID) should it be?